services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/resource/S3ArnUtils.java

```diff
@@ -72,4 +72,14 @@ public static IntermediateOutpostResource parseOutpostArn(Arn arn) {
                                           .outpostSubresource(ArnResource.fromString(subresource))
                                           .build();
     }
+
+    public static boolean isArnFor(S3ResourceType s3ResourceType, String arnString) {
```

Comment:
Nit: Should this be getArnType so that we don't have to parse the ARN twice?

Reply:
Good idea. Especially since we're relying on exception handling here, which is not the most performant to begin with.